pipelines/curate_rewards.py

```
@@ -0,0 +1,917 @@
+):
+    """Convert one JSONL file with no-clobber output and sidecar destinations."""
+    input_path = Path(input_path)
+    output_path = Path(output_path)
+    sidecar_path = Path(sidecar_path)
+    if input_path.resolve() in {output_path.resolve(), sidecar_path.resolve()}:
+        raise RewardOntologyError("input and output paths must be distinct")
+    if output_path.resolve() == sidecar_path.resolve():
+        raise RewardOntologyError("record and sidecar outputs must be distinct")
+    for destination in (output_path, sidecar_path):
+        if destination.exists():
+            raise RewardOntologyError(f"refusing to overwrite existing path: {destination}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)